tree/dataframe/test/dataframe_merge_results.cxx

```diff
@@ -0,0 +1,172 @@
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RResultPtr.hxx"
+#include "gtest/gtest.h"
+
+TEST(RDataFrameMergeResults, MergeCount)
+{
+    ROOT::RDataFrame df{100};
+    auto mycount = df.Count();
+
+    auto mergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+    auto othermergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+
+    mergeablecount->Merge(othermergeablecount);
+    auto mergedcount = mergeablecount->GetValue();
+
+    EXPECT_EQ(mergedcount, 200);
+}
+
+TEST(RDataFrameMergeResults, MergeMean)
+{
+    ROOT::RDataFrame df1{10};
+    ROOT::RDataFrame df2{20};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto mean1 = col1.Mean("x");
+    auto mean2 = col2.Mean("x");
+
+    auto mergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean1);
+    auto othermergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean2);
+
+    mergeablemean->Merge(othermergeablemean);
+
+    auto mergedmean = mergeablemean->GetValue();
+
+    EXPECT_NEAR(mergedmean, 7.83333333333, 0.01);
```

Comment:
we should be able to guarantee higher precision than this?

Reply:
Yes actually, this being a simple test I wasn't paying too much attention to precision

Reply:
Changed to EXPECT_DOUBLE in the tests